### PR TITLE
NEXUS-12701 Update bower-nexus3-resolver dependencies

### PR DESCRIPTION
**Pull request by fjmilens3**
General update of dependencies triggered by https://github.com/sonatype/bower-nexus3-resolver/pull/13. 

(Waiting for internal Sonatype testing prior to merging.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,16 +11,16 @@
     "src"
   ],
   "dependencies": {
-    "q": "~1.4.1",
-    "request": "~2.67.0",
-    "tar": "~2.1.1",
-    "tmp": "~0.0.28"
+    "q": "~1.5.0",
```
Comment (jlstephens89) [inline package.json]: I'm assuming it was a conscious effort to not go for v2 and above?

Comment (fjmilens3) [inline package.json]: I believe `1.5.0` is the latest: https://www.npmjs.com/package/q

Comment (jlstephens89) [inline package.json]: To complete the circle it turns out 2.0 and above is experimental atm!

File: package.json [hunk 2]
```diff
@@ -11,16 +11,16 @@
...
+    "tmp": "~0.0.31"
   },
   "devDependencies": {
-    "mocha": "~2.3.4",
-    "mocha-junit-reporter": "~1.9.1",
-    "nock": "~5.2.1",
-    "sinon": "~1.17.2"
+    "mocha": "~3.2.0",
+    "mocha-junit-reporter": "~1.13.0",
+    "nock": "~9.0.9",
```
Comment (DarthHater) [inline package.json]: This version is not supported on node 5.x (which isn't supported, so probably fine). According to nock that version supports up to nock 8.x

Comment (fjmilens3) [inline package.json]: Good to know. Also as one of the `devDependencies` I'm assuming this shouldn't hurt people actually _using_ the resolver, just that they'd need to be on something other than 5.x if they want to work on it.

**Pull request merged**